packages/core/src/provider/base.ts

```diff
@@ -49,48 +49,19 @@ import type {
   PaymasterAndDataMiddleware,
   ProviderEvents,
   SendUserOperationResult,
+  SmartAccountProviderConfig,
 } from "./types.js";
+import { SmartAccountProviderConfigSchema } from "./validation.js";
 
 export const noOpMiddleware: AccountMiddlewareFn = async (
   struct: Deferrable<UserOperationStruct>
 ) => struct;
 
-export interface SmartAccountProviderOpts {
```

Comment:
we should export these still as export them as `z.infer<typeof SmartAccountProviderOptsSchema>`